src/include/homestore/btree/detail/varlen_node.hpp

```diff
@@ -63,10 +71,6 @@ class VariableNode : public BtreeNode {
 
     virtual ~VariableNode() = default;
 
-    uint32_t occupied_size(const BtreeConfig& cfg) const override {
-        return (cfg.node_data_size() - sizeof(var_node_header) - available_size(cfg));
```

Comment:
It is overriden here because in merge_nodes(), for varlen node type  `available_size -= old_nodes[i]->occupied_size();` results in miscalculation. It can happen frequently with reduced node size (for instance 512). 
In merge if the total entries of a node is needed to move to left, only the size of actual data matters not `var_node_header`.

Reply:
I am probably missing something, but doing that overridden would defeat the generic purpose right? I am not clear why available_size() doesn't take discount the `var_node_header` area?